pkg/terra/contract_cache.go

```diff
@@ -0,0 +1,426 @@
+	}
+
+	return details.LatestConfigDigest, details.Epoch, details.Round, ans, time.Unix(details.LatestTimestamp, 0), nil
+}
+
+func (cc *ContractCache) latestTransmissionDetails() (
+	configDigest types.ConfigDigest,
+	epoch uint32,
+	round uint8,
+	latestAnswer *big.Int,
+	latestTimestamp time.Time,
+	err error,
+) {
+	cc.transMu.RLock()
+	ts := cc.transTS
+	configDigest = cc.digest
+	epoch = cc.epoch
+	round = cc.round
+	latestAnswer = cc.latestAnswer
+	latestTimestamp = cc.latestTimestamp
+	cc.transMu.RUnlock()
+	err = cc.checkTS(ts)
+	return
+}
+
+// fetchLatestRoundRequested fetches the latest round requested by filtering event logs
+//func (cc *ContractCache) fetchLatestRoundRequested(ctx context.Context, lookback time.Duration) (
+//	configDigest types.ConfigDigest,
+//	epoch uint32,
+//	round uint8,
+//	err error,
+//) {
+//	// calculate start block
+//	latestBlock, blkErr := cc.chainReader.LatestBlock()
+//	if blkErr != nil {
+//		err = blkErr
+//		return
+//	}
+//	blockNum := uint64(latestBlock.Block.Header.Height) - uint64(lookback/cc.cfg.BlockRate())
+//	res, err := cc.chainReader.TxsEvents([]string{fmt.Sprintf("tx.height>=%d", blockNum+1), fmt.Sprintf("wasm-new_round.contract_address='%s'", cc.address.String())})
+//	if err != nil {
+//		return
+//	}
+//	if len(res.TxResponses) == 0 {
+//		return
+//	}
+//	if len(res.TxResponses[0].Logs) == 0 {
+//		err = fmt.Errorf("No logs found for tx %s", res.TxResponses[0].TxHash)
+//		return
+//	}
+//	// First tx is the latest.
+//	if len(res.TxResponses[0].Logs[0].Events) == 0 {
+//		err = fmt.Errorf("No events found for tx %s", res.TxResponses[0].TxHash)
+//		return
+//	}
+//
+//	for _, event := range res.TxResponses[0].Logs[0].Events {
+//		if event.Type == "wasm-new_round" {
+//			// TODO: confirm event parameters
+//			// https://github.com/smartcontractkit/chainlink-terra/issues/22
+//			for _, attr := range event.Attributes {
+//				key, value := string(attr.Key), string(attr.Value)
+//				switch key {
+//				case "latest_config_digest":
+//					// parse byte array encoded as hex string
+//					if err := HexToConfigDigest(value, &configDigest); err != nil {
+//						return configDigest, epoch, round, err
+//					}
+//				case "epoch":
+//					epochU64, err := strconv.ParseUint(value, 10, 32)
+//					if err != nil {
+//						return configDigest, epoch, round, err
+//					}
+//					epoch = uint32(epochU64)
+//				case "round":
+//					roundU64, err := strconv.ParseUint(value, 10, 8)
+//					if err != nil {
+//						return configDigest, epoch, round, err
+//					}
+//					round = uint8(roundU64)
+//				}
+//			}
+//			return // exit once all parameters are processed
+//		}
+//	}
+//	return
+//}
+
+func (cc *ContractCache) latestRoundRequested(lookback time.Duration) (
+	configDigest types.ConfigDigest,
+	epoch uint32,
+	round uint8,
+	err error,
+) {
+	cc.transMu.RLock()
+	ts := cc.transTS
+	configDigest = cc.digest
+	epoch = cc.epoch
+	round = cc.round
```

Comment:
The Solana version returns `0`s here for epoch and round. Should we do the same?

Reply:
no, that actually seems like a bug in the solana version or possibly they are intentionally not supporting the explicit round requesting for some reason

Reply:
@aalu1418  https://github.com/smartcontractkit/chainlink-solana/blob/develop/pkg/solana/median_contract.go#L59

Reply:
yeah turns out for solana the rounds are fast enough that they reasoned to just disable this feature essentially